### PR TITLE
fix(cli): add impersonate_user to db import

### PR DESCRIPTION
**Pull request by chessman**
### SUMMARY
When importing a directory using `superset import-directory` command, it should respect the `impersonate_user` flag specified in the database yaml definition.

### TESTING INSTRUCTIONS
Import a database that supports this flag and verify that this flag is enabled.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by mistercrunch — APPROVED** [review]
Makes me wonder whether other fields may get lost while doing export/import round trips.

**Comment by rusackas** [comment]
I believe @yousoph and @eschutho are taking stock of bits that are not part of import/export but ought to be.

File: superset/models/core.py [hunk 1]
```diff
@@ -171,6 +171,7 @@ class Database(Model, AuditMixinNullable, ImportExportMixin):  # pylint: disable
         "allow_dml",
         "allow_file_upload",
         "extra",
+        "impersonate_user"
```
Comment (eschutho) [inline superset/models/core.py:174]:
```suggestion
        "impersonate_user",
```

**Comment by eschutho** [comment]
Can we add a unit test with the new param? Here's an example in `tests/unit_tests/databases/commands/importers/v1/import_test.py`

```
def test_import_database_with_user_impersonation(
    mocker: MockerFixture,
    session: Session,
) -> None:
    """
    Test importing a database that is managed externally.
    """
    from superset import security_manager
    from superset.commands.database.importers.v1.utils import import_database
    from superset.models.core import Database
    from tests.integration_tests.fixtures.importexport import database_config

    mocker.patch.object(security_manager, "can_access", return_value=True)

    engine = db.session.get_bind()
    Database.metadata.create_all(engine)  # pylint: disable=no-member

    config = copy.deepcopy(database_config)
    config["impersonate_user"] = True

    database = import_database(config)
    assert database.impersonate_user is True
    ```

**Comment by rusackas** [comment]
Approving CI. Ping me if it passes and needs a merge!

**Comment by chessman** [comment]
@rusackas This error looks unrelated to my changes.
```
>       conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
E       sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "host1" to address: Temporary failure in name resolution
E       
E       (Background on this error at: https://sqlalche.me/e/14/e3q8)
```

File: tests/unit_tests/databases/commands/importers/v1/import_test.py [hunk 2]
```diff
@@ -172,6 +172,30 @@ def test_import_database_with_version(mocker: MockerFixture, session: Session) -
...
+    """
+    Test importing a database that is managed externally.
+    """
+    from superset import security_manager
+    from superset.commands.database.importers.v1.utils import import_database
+    from superset.models.core import Database
+    from tests.integration_tests.fixtures.importexport import database_config
+
+    mocker.patch.object(security_manager, "can_access", return_value=True)
+
```
Comment (eschutho) [inline tests/unit_tests/databases/commands/importers/v1/import_test.py:188]:
```suggestion
mocker.patch("superset.commands.database.importers.v1.utils.add_permissions")
```

**Comment by rusackas** [comment]
Running CI. 🤞 Ping me if it turns green :D

**Comment by chessman** [comment]
@rusackas it's green!

**Comment by eschutho** [comment]
Thanks @chessman!

**Pull request merged**